swift/ql/lib/codeql/swift/frameworks/StandardLibrary/CustomUrlSchemes.qll

```diff
@@ -0,0 +1,33 @@
+import swift
+private import codeql.swift.dataflow.DataFlow
+private import codeql.swift.dataflow.FlowSources
+private import codeql.swift.dataflow.ExternalFlow
+
+private class CustomUrlSchemeRemoteFlowSource extends RemoteFlowSource, DataFlow::ParameterNode {
+  CustomUrlSchemeRemoteFlowSource() {
+    exists(AbstractFunctionDecl f |
+      f.getName() =
+        [
+          "application(_:open:options:)", "application(_:handleOpen:)",
+          "application(_:open:sourceApplication:annotation:)",
+        ] and
+      f.getEnclosingDecl().(ClassDecl).getABaseType().getName() = "UIApplicationDelegate"
```

Comment:
Sorry, that was leftover code from an experiment I did before using MaD rows 🤦‍♂️.

MaD is enough to cover these three cases, since we know the position of the `URL` parameter in each call.

Reply:
Ah, fantastic! I was also very sure that I had implemented MaD for parameters at some point (but not sure enough to write that in my review!) 😂